esp8266/esp_mphal.c

```diff
@@ -146,6 +149,31 @@ void mp_hal_signal_input(void) {
     #endif
 }
 
+void mp_hal_uart_rx_intr(int uart_no) {
```

Comment:
IMO this belongs in machine_uart.c, since it's specific to UART behaviour.  Also, since UART1 can't receive it doesn't need to take an argument.